### DISCUSSION TITLE
Manual installation of JavaHL for Eclipse on Windows

### DISCUSSION THREAD

**Discussion by mooserator** [subclipse/subclipse#213] [Q&A] [answered]
We want to install Subclipse on VMs in a data center that is blocking both the Eclipse Update sites and Marketplace with a firewall.  Getting the firewall rules changed is essentially impossible, leaving us to download the packages and install them manually within Eclipse.  We were able to do this back in 2019, locating appropriate files for Subclipse, SVNKit, and JavaHL.

Trying to use the updated file versions this time.  No problems with the SVNKit (org.tmatesoft.svn_1.10.3.eclipse.zip) or Subclipse (subclipse-4.3.3.zip).  Both of these can be manually installed without issue, provided the SVNKit is installed before Subclipse.

The problem is with JavaHL.  We have yet to find a 1.14.x zip package that can be installed successfully through Eclipse.  Have tried several iterations of javahl-windows-1.14.0.zip and javahl-1.14.0.zip.  Eclipse won't install anything from these, returning an error message that says that it could not find the jar within the zip file.  Is there a way to manually install Subclipse JavaHL in Eclipse?  Any help would be greatly appreciated.

![javahl-err-screenshot](https://user-images.githubusercontent.com/78229331/149028465-e6674e60-5d02-4f5c-87a9-2cdad45e8af8.JPG)

**Comment by markphip** [comment, marked as answer]
I only examined the contents of the zip file, but this file looks like it would be correct:

https://github.com/subclipse/updates/blob/main/javahl/subclipse-javahl-1.14.0.zip

**Reply by mooserator** [reply]
Thank you Mark!  That appears to be exactly what we needed.  JavaHL is now present and we are beginning to checkout code from the repository.  Much appreciated!  :)

**Reply by markphip** [reply]
Thanks for confirming. I added the zip to the Subclipse release to make it easier in the future for people to find it:  https://github.com/subclipse/subclipse/releases/tag/4.3.3
